src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
Because `charset` is optional ("may be specified" are words from RFC), you'll convert a proper text into a byte array from time to time. It would be better to specify a body return type upon request, rather then try to convert it based on a `Context-Type` parameters.